### PR TITLE
Chore: Replace PR with truncated commit hash when build fails

### LINKED ISSUE #28059: Add PR name and author to Drone notifications
**What would you like to be added**:

PR name and author information to the Drone notifications

**Why is this needed**:

The current notifications are good to have but I'm not interested in all build failures, it would be great to see a notification and know if I need to care (it is something I'm involved in).

### PR DESCRIPTION
**Pull request by dsotirakis**
**What this PR does / why we need it**:

After #31124 was done, we noticed that when a pipeline runs against master, the PR number doesn't exist anymore.

Probably, the most appropriate way to use it, is to include the commit that causes this pipeline to fail. Moreover, this may be beneficial when squashing the commit - it will show the aggregated changes.

Essentially, `<https://github.com/{{repo.owner}}/{{repo.name}}/commit/{{build.commit}}|{{ truncate build.commit 8 }}>` will produce a link to the 8-character commit hash, so it can be clickable and reviewable.

**Which issue(s) this PR fixes**:

Fixes #28059

### CODE REVIEW

**Review by aknuds1 — APPROVED** [review]
Pretty much looks good, but see suggestion.

File: scripts/lib.star [hunk 1]
```diff
@@ -71,7 +71,7 @@ def slack_step(channel):
                 'from_secret': 'slack_webhook',
             },
             'channel': channel,
-            'template': 'Build {{build.number}} failed for PR {{build.pull}}: {{build.link}}\nAuthor: {{build.author}}',
+            'template': 'Build {{build.number}} failed for Commit: <https://github.com/{{repo.owner}}/{{repo.name}}/commit/{{build.commit}}|{{ truncate build.commit 8 }}>: {{build.link}}\nAuthor: {{build.author}}',
```
Comment (aknuds1) [inline scripts/lib.star:74]: Commit shouldn't be capitalized. Why is the URL in angle brackets?
```suggestion
            'template': 'Build {{build.number}} failed for commit: <https://github.com/{{repo.owner}}/{{repo.name}}/commit/{{build.commit}}|{{ truncate build.commit 8 }}>: {{build.link}}\nAuthor: {{build.author}}',
```

**Review by aknuds1 — APPROVED** [review]

File: scripts/lib.star (hunk 1 above)
Comment (dsotirakis) [reply scripts/lib.star]: It's in angle brackets so it can produce a clickable link on `.drone.yml`

**Review by marefr — COMMENTED** [review]
LGTM lets try it out and see how it works

**Pull request merged**